x/protocolpool/simulation/operations.go

```diff
@@ -0,0 +1,88 @@
+package simulation
+
+import (
+	"math/rand"
+
+	"cosmossdk.io/math"
+
+	"github.com/cosmos/cosmos-sdk/baseapp"
+	"github.com/cosmos/cosmos-sdk/client"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	simtypes "github.com/cosmos/cosmos-sdk/types/simulation"
+	"github.com/cosmos/cosmos-sdk/x/protocolpool/keeper"
+	"github.com/cosmos/cosmos-sdk/x/protocolpool/types"
+	"github.com/cosmos/cosmos-sdk/x/simulation"
+)
+
+var TypeFundCommunityPool = sdk.MsgTypeURL(&types.MsgFundCommunityPool{})
+
+// Simulation operation weights constants
+const (
+	OpWeightMsgFundCommunityPool = "op_weight_msg_fund_community_pool"
+
+	DefaultWeightMsgFundCommunityPool = 100
+)
+
+// WeightedOperations returns all the operations from the module with their respective weights
+func WeightedOperations(
+	appParams simtypes.AppParams,
+	txGen client.TxConfig,
+	ak types.AccountKeeper,
+	bk types.BankKeeper,
+	k keeper.Keeper,
+) simulation.WeightedOperations {
+	var weightMsgFundCommunityPool int
+
+	appParams.GetOrGenerate(OpWeightMsgFundCommunityPool, &weightMsgFundCommunityPool, nil,
+		func(_ *rand.Rand) {
+			weightMsgFundCommunityPool = DefaultWeightMsgFundCommunityPool
+		},
+	)
+
+	return simulation.WeightedOperations{
+		simulation.NewWeightedOperation(
+			weightMsgFundCommunityPool,
+			SimulateMsgFundCommunityPool(txGen, ak, bk, k),
+		),
+	}
+}
+
+func SimulateMsgFundCommunityPool(
+	txGen client.TxConfig,
+	ak types.AccountKeeper,
+	bk types.BankKeeper,
+	_ keeper.Keeper,
+) simtypes.Operation {
+	return func(r *rand.Rand, app *baseapp.BaseApp, ctx sdk.Context, accs []simtypes.Account, chainID string,
+	) (simtypes.OperationMsg, []simtypes.FutureOperation, error) {
+		simAccount, _ := simtypes.RandomAcc(r, accs)
+		account := ak.GetAccount(ctx, simAccount.Address)
+		spendable := bk.SpendableCoins(ctx, account.GetAddress())
+		// choose 50% of spendable
+		spendableSubAmount := types.PercentageCoinMul(math.LegacyMustNewDecFromStr("0.5"), spendable)
+		if spendableSubAmount.IsZero() {
+			return simtypes.NoOpMsg(types.ModuleName, TypeFundCommunityPool, "no balance"), nil, nil
```

Comment:
why 50%?

Reply:
just some arbitrary amount that is not "full balance" so we have enough to pay fees

Reply:
we could use rand here to get something between 10-70 or some upper bound?